src/examples/libpmem2/pmemset/main.c

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+#include "pmemset.h"
+#include <stdio.h>
+#include <assert.h>
+
+int
+main(int argc, char *argv[])
+{
+	struct pmemset_config *config;
+	pmemset_config_new(&config);
+	pmemset_config_set_create_if_none(config, 1);
+
+	struct pmemset_source *source;
+	pmemset_source_from_file(&source, "/mnt/pmem/testfile");
+
+	struct pmemset *set;
+	pmemset_new(&set, config);
+
+	struct pmemset_part_descriptor *descriptor;
+	pmemset_part_descriptor_new(&descriptor, set, source, 0, 1 << 20);
+
+	enum pmemset_part_state state;
+	pmemset_part_finalize(set, descriptor, NULL, NULL, NULL, NULL, &state);
```

Comment:
Null null null null 

winapi style ;)

Reply:
You would typically use those parameters. This is just the simplest "working" example of creeating a set.